File: src/main/java/com/finchool/server/entities/Achievement.java
package com.finchool.server.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "achievement")
public class Achievement extends BaseEntity{
    private String name;
    private List<User> users = new ArrayList<>();

    protected Achievement(){}

    public Achievement(String name, List<User> users) {
        this.name = name;
        this.users = users;
    }

    @Column(nullable = false, unique = true, length = 250, name = "name")
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @ManyToMany(mappedBy = "achievementsReceived")
    public List<User> getUsers() {
        return users;
    }

    public void setUsers(List<User> users) {
        this.users = users;
    }
}

File: src/main/kotlin/com/finschool/Server/models/Achievement.kt
package com.finschool.Server.models


import jakarta.persistence.*

@Entity
@Table(name = "Achievement")
class Achievement (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, unique = true, name = "AchId")
    val id: Int? = null,

    @Column(nullable = false, unique = true, length = 250, name = "AchName")
    var name: String,

    @ManyToMany(mappedBy = "achievementsReceived", fetch = FetchType.EAGER)
    val users: MutableList<User> = mutableListOf()
)